exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -924,11 +924,85 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
      * when using the PostgreSQL dialect is allowed, but this value will be ignored by the database.
      * @throws IllegalStateException If no column type mapping is found.
      */
-    inline fun <reified E : Any> array(name: String, maximumCardinality: Int? = null): Column<List<E>> {
+    inline fun <reified E : Any> array(name: String, maximumCardinality: Int? = null): Column<List<E>> =
+        arrayN<E, List<E>>(name, dimensions = 1, maximumCardinality?.let { listOf(it) })
+
+    /**
+     * Creates a 3-dimensional array column, with the specified [name], for storing elements of a nested `List`.
+     *
+     * **Note:** This column type is only supported by PostgreSQL dialect.
+     *
+     * @param name Name of the column.
+     * @param maximumCardinality The maximum cardinality (number of allowed elements) for each dimension in the array.
+     *
+     * **Note:** Providing an array size limit when using the PostgreSQL dialect is allowed, but this value will be ignored by the database.
+     * The whole validation is performed on the client side.
+     *
+     * @return A column instance that represents a 3-dimensional list of elements of type [T].
+     * @throws IllegalStateException If no column type mapping is found.
+     */
+    inline fun <reified T : Any> Table.array3(name: String, maximumCardinality: List<Int>? = null): Column<List<List<List<T>>>> =
+        arrayN<T, List<List<List<T>>>>(name, dimensions = 3, maximumCardinality)
+
+    /**
+     * Creates a 2-dimensional array column, with the specified [name], for storing elements of a nested `List`.
+     *
+     * **Note:** This column type is only supported by PostgreSQL dialect.
+     *
+     * @param name Name of the column.
+     * @param maximumCardinality The maximum cardinality (number of allowed elements) for each dimension in the array.
+     *
+     * **Note:** Providing an array size limit when using the PostgreSQL dialect is allowed, but this value will be ignored by the database.
+     * The whole validation is performed on the client side.
+     *
+     * @return A column instance that represents a 2-dimensional list of elements of type [T].
+     * @throws IllegalStateException If no column type mapping is found.
+     */
+    inline fun <reified T : Any> Table.array2(name: String, maximumCardinality: List<Int>? = null): Column<List<List<T>>> =
+        arrayN<T, List<List<T>>>(name, dimensions = 2, maximumCardinality)
```

Comment:
As a variant, we can remove `array2` and `array3` from API for now, and keep only 2 versions of `array` (for 1x and Nx arrays). It's not so popular feature that is needed for everyone, and those who use it can create such util functions by themselves.